### PR TITLE
Output hash of pubkeys and validation bits from batch verification circuit

### PR DESCRIPTION
**Pull request by rahulghangas**
Batch verification will now have three output signals

1. two signals of the hash `hash[2]`, where the the first and second signal represent the first 128 bits and the latter 128 bits from the output of `keccak256`(ethereum compatible)
2. one `verified` signal, wherein the bits (from lsb to msb) will represent whether the signature validation for a particular known publickey passed or not

### CODE REVIEW

File: circuits/batchverify.circom [hunk 1]
```diff
@@ -12,29 +17,58 @@ template BatchVerify(n, m) {
...
+      verifiers[i].msg[j] <== msg[j];
     }
     for (j=0; j<255; j++) {
-      verifiers[i].A[j] = A[i][j];
-      verifiers[i].R8[j] = R8[i][j];
-      verifiers[i].S[j] = S[i][j];
+      verifiers[i].A[j] <== A[i][j];
+      verifiers[i].R8[j] <== R8[i][j];
+      verifiers[i].S[j] <== S[i][j];
     }
```
Comment (jinankjain) [inline circuits/batchverify.circom:41]: Wasn't this already fixed in #50?

File: circuits/verify.circom [hunk 2]
```diff
@@ -10,6 +10,8 @@ include "../node_modules/circomlib/circuits/comparators.circom";
 include "../node_modules/circomlib/circuits/gates.circom";
 
 template Ed25519Verifier(n) {
+  assert(n\8 * 8 == n);
```
Comment (jinankjain) [inline circuits/verify.circom:13]: nit: assert(n % 8 == 0);

**Review by jinankjain — APPROVED** [review]
I think some changes are already merged to master, so may just rebase. Apart from that PR looks good to me.

File: circuits/verify.circom (hunk 2 above)
Comment (rahulghangas) [reply circuits/verify.circom:13]: Oh yup, that was dumb on my part. Sorry

File: circuits/batchverify.circom (hunk 1 above)
Comment (rahulghangas) [reply circuits/batchverify.circom:41]: Correct, will rebase

**Pull request merged**